benchmark/buffers/buffer-bytelength.js

```diff
@@ -0,0 +1,36 @@
+var common = require('../common');
+
+var bench = common.createBenchmark(main, {
+  encoding: ['utf8', 'base64'],
+  len: [4, 16, 64, 256, 1024],
+  n: [1e7]
+});
+
+// 16 chars of each byte length
+var encodings = {
+  'utf8': {
+    0: 'hello brendan!!!', // 1 byte
+    1: 'ΰαβγδεζηθικλμνξο', // 2 bytes
+    2: '挰挱挲挳挴挵挶挷挸挹挺挻挼挽挾挿', // 3 bytes
+    3: '𠜎𠜱𠝹𠱓𠱸𠲖𠳏𠳕𠴕𠵼𠵿𠸎𠸏𠹷𠺝𠺢' // 4 bytes
+  },
+  'base64': {
+    0: 'aGVsbG8gd29ybGQh', // no padding
+    1: 'YW55IGNhcm5hbCBwbGVhc3VyZS4=', // one byte
+    2: 'YWJjZGVmZ2hpamtsbW5eYw==', // two bytes
+    3: 'YXN1cmUu', // nada
+  }
```

Comment:
The strings are perhaps too short to really stress-test the function.  It would be good to have some long strings and some deeply nested cons strings.